runelite-client/src/main/java/net/runelite/client/plugins/inventorytagger/InventoryTaggerPlugin.java

```diff
@@ -0,0 +1,183 @@
+package net.runelite.client.plugins.inventorytagger;
+
+import com.google.common.eventbus.Subscribe;
+import java.awt.Color;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuOpened;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.menus.MenuManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.util.Text;
+import org.apache.commons.lang3.ArrayUtils;
+
+@PluginDescriptor(
+	name = "Inventory Item Tagging",
+	enabledByDefault = true
+)
+public class InventoryTaggerPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private InventoryTaggerPlugin plugin;
+
+	@Inject
+	private InventoryTaggerOverlay overlay;
+
+	public Map<String, TaggedItems> taggedItems = new HashMap<>();
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	private boolean editorMode = false;
+
+	private static final String SETNAME_MELE = "MELE";
```

Comment:
This should probably be "melee". I personally think it would look better written as "Melee" and "Range" etc. as well.